test/parallel/test-typescript.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+const { spawnPromisified } = require('../common');
+const fixtures = require('../common/fixtures');
+const { strictEqual } = require('node:assert');
+const { test } = require('node:test');
+
+test('execute a typescript file', async () => {
+  const {
+    code,
+    stderr,
+    stdout,
+  } = await spawnPromisified(process.execPath, [
+    '--experimental-typescript',
+    fixtures.path('typescript/test-typescript.ts'),
+  ], { stdio: ['inherit'] });
+
+  strictEqual(code, 0);
+  strictEqual(stderr, '');
+  strictEqual(stdout, 'Hello, TypeScript!\n');
```

Comment:
Putting `code` first is an anti-pattern, when the test fails it throws a very non-informative error about `1` not being equal to `0`. You should either put `stderr` first, then `stdout`, and finally `code` – or better use an object
```suggestion
  deepStrictEqual(result, {
    code: 0,
    signal: null,
    stderr: '',
    stdout: 'Hello, TypeScript!\n',
  });
```